pkg/controller/controller.go

```diff
@@ -243,18 +244,20 @@ func (c *Controller) monitor(watchVersion string) (<-chan *Event, <-chan error)
 				if err != nil {
 					if err == io.EOF { // apiserver will close stream periodically
 						c.logger.Debug("Apiserver closed stream")
+						// go to the next round
 						break
 					}
 
 					c.logger.Errorf("Received invalid event from API server: %v", err)
-					errCh <- err
-					return
+					// keep polling next event
+					continue
```

Comment:
Consider adding loop labels to make it clear whether you intend to jump to the per-message loop or outer per-watch loop (here and on the other nearby `continue` and `break` statements).